clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -950,6 +950,45 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (getLangOpts().SYCLIsHost && D && D->hasAttr<SYCLKernelAttr>())
     Fn->addFnAttr("sycl_kernel");
 
+  if (getLangOpts().SYCL && D && (D->hasAttr<SYCLIntelLoopFuseAttr>())) {
+    auto *A = D->getAttr<SYCLIntelLoopFuseAttr>();
+    Expr *E = A->getValue();
+
+    // Emit '1' if optional argument is omitted.
+    llvm::ConstantInt *Value = Builder.getInt32(1);
+
+    // Emit argument if specified.
+    if (E)
+      Value = Builder.getInt32(
+          E->getIntegerConstantExpr(D->getASTContext())->getSExtValue());
+
+    llvm::Metadata *AttrMDArgs[] = {
+        llvm::ConstantAsMetadata::get(Value),
+        llvm::ConstantAsMetadata::get(Builder.getInt32(0))};
+    Fn->setMetadata("loop_fuse",
+                    llvm::MDNode::get(getLLVMContext(), AttrMDArgs));
+  }
+
+  if (getLangOpts().SYCL && D &&
+      (D->hasAttr<SYCLIntelLoopFuseIndependentAttr>())) {
+    auto *A = D->getAttr<SYCLIntelLoopFuseIndependentAttr>();
+    Expr *E = A->getValue();
+
+    // Emit '1' if optional argument is omitted.
+    llvm::ConstantInt *Value = Builder.getInt32(1);
+
+    // Emit argument if specified.
+    if (E)
+      Value = Builder.getInt32(
+          E->getIntegerConstantExpr(D->getASTContext())->getSExtValue());
+
+    llvm::Metadata *AttrMDArgs[] = {
+        llvm::ConstantAsMetadata::get(Value),
+        llvm::ConstantAsMetadata::get(Builder.getInt32(1))};
```

Comment:
There's a fair amount of code duplication here -- it took me a while to discover that the only difference between these two blocks is the `1` here. Perhaps this should be factored into a lambda to reduce code duplication and make the distinction more clear?